tests/test_io.py

```
@@ -105,3 +105,33 @@ def __repr__(self):
+
+    # apply_wildcards() is the correct, flag-preserving way to do this -- confirm it actually
+    # works as the baseline this test is protecting.
+    substituted = flagged.apply_wildcards({"sample": "foo"})
+    assert substituted == "results/foo.txt"
+    assert substituted.is_temp
+
+    # .format() must refuse rather than silently return an unflagged (and not even _IOFile)
+    # result.
+    try:
+        flagged.format(sample="foo")
+        assert False, ".format() on a flagged _IOFile should raise WorkflowError"
```

Ruff (0.16.1)

[warning] Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)